azavea/pfb-network-connectivity
# Add Neighborhood.Visibility.HIDDEN option

Additionally hide HIDDEN neighborhoods from geojson



## Overview

Adds an additional visibility option, that hides a neighborhood from all public views on the site.

This allows us to set the MN neighborhoods/jobs to HIDDEN on production after this PR is deployed with the final site changes, and make them reappear later if PFB desires for any reason.

### Demo

Not so easy to demo, see testing section below.


### Notes

_Copied from commit message:_

Ideally, we should default to only returning PUBLIC.
Then if the request is authenticated, also return PRIVATE
neighborhoods. This is more complicated, though, and I
think I've exhausted my "make an incremental improvement
by spending a bit more time" budget by adding the HIDDEN
visibility option rather than just deleting all MN
jobs from the production site to address #541.

The private neighborhoods appearing on the home page
"is your place on the map?" map was already an issue here,
so we're not making things worse.

If this is deemed unnecessary or there are complications from adding HIDDEN that arise during code review, we should be ok to simply close this PR and manually delete the MN jobs from production.

## Testing Instructions

- Open a django shell console
- Get a ref to a neighborhood that has a job with `status=COMPLETE`
- Modify the neighborhood's visibility to `PRIVATE`, then refresh the site.
  - If you're signed out, the neighborhood should no longer be visible, except on the "is your place on the map?" map.
  - If you're signed in, the neighborhood should remain visible
- Modify the neighborhood's visibility to `HIDDEN`, then refresh the site.
  - If you're signed out, the neighborhood should no longer be visible, except on the "is your place on the map?" map.
  - If you're signed in, the neighborhood should no longer be visible, except on the "is your place on the map?" map.


Connects #541
